### PR TITLE
Make integration tests use demo keys from testonly package

### PR DESCRIPTION
**Pull request by RJPercival**
Rather than loading keys from testdata/log-rpc-server-{pub,priv}key.pem files. 

Finishes what PR #678 started.

### CODE REVIEW

File: testonly/integration/clientserver.go [hunk 1]
```diff
@@ -46,13 +46,25 @@ var (
...
-	privateKeyInfo    = &keyspb.PEMKeyFile{
-		Path:     testonly.RelativeToPackage("../../testdata/log-rpc-server.privkey.pem"),
-		Password: "towel",
+	publicKey         = testonly.DemoPublicKey
+	privateKeyInfo    = &keyspb.PrivateKey{
+		Der: privatePEMToDER(testonly.DemoPrivateKey, testonly.DemoPrivateKeyPass),
 	}
 )
 
+func privatePEMToDER(pem, password string) []byte {
```
Comment (codingllama) [inline testonly/integration/clientserver.go]: Is this the same logic as the one on admin_storage_tester.go? Could we reuse it? (We can do it in a separate PR if you prefer.)

Comment (RJPercival) [inline testonly/integration/clientserver.go]: It is the same logic, yes. I'll see if I can find a common place for it to live in a separate PR.
